### PR TITLE
Comment Moderation Function using Machine Learning with Nyckel

### LINKED ISSUE #6085: [Machine Learning] Create a comment moderation function using machine learning from Nyckel
#### Proposed article introduction
Nyckel is Machine Learning API which has proven very efficient in automated model training even for those with little knowledge in machine learning. A comment function identifies and classify messages with reference to the content it holds. This article creates a comment function, trains a model using Nyckel and integrates the Nyckel model with the function to classify messages on likeliness of spam.
#### Key takeaways 
- Understanding Nyckel API
- Create and train the model using Nyckel.
- Create the Comment function
- Invoke the model from the function
- Model testing
#### Article quality
Unlike other articles, this article trains a model using Nyckel API and invokes it from the comment function. It will
be easy to understand for beginners since the model creation and training using Nyckel is very easy.

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by del-ui**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6085 
> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by LinusMuema** [comment]
plagiarism check done ✔️

**Comment by M0nicah** [comment]
hello @del-ui,
is it `comment moderation` or `content moderation` ?

**Review by M0nicah — CHANGES_REQUESTED** [review]

File: content/articles/comment-moderation-function-using-machine-learning-from-nyckel/index.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+![home](engineering-education/comment-moderation-using-machine-learning-from-nyckel/home.png)
+
+### Creating the comment function
+Once you are on the home page, click the `New function` button at the bottom of the page, after which you will be directed to the page shown below:
+
+![function](engineering-education/comment-moderation-using-machine-learning-from-nyckel/function.png)
+
+This page gives you two types of data input i.e. text and image. In this case, you are classifying comments, so the text will be the preferable input.
+
+Select _text_ then add the labels in the forms that follow as `spam` and `Not spam`. Afterwards, hit the **create function** button which will create the function named **NewFunction**. You can rename this function name to give a name like `commentFunction` for easy identification in case you have
```
[the remark this answers is not in the document]
Comment (del-ui) [reply content/articles/comment-moderation-function-using-machine-learning-from-nyckel/index.md]: @M0nicah  I've got it in place

**Review by M0nicah — APPROVED** [review]
🚀 Awesome article @del-ui  , please review the changes I made in the commits to help improve the flow and format for your future articles.
Thanks!

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@del-ui 

Victor can you please DM (via Slack) asap 👍

**Pull request merged**